compiler/src/iree/compiler/Utils/TracingUtils.cpp

```diff
@@ -51,11 +51,24 @@ void *operator new[](std::size_t count) {
   return ptr;
 }
 void *operator new(std::size_t count, std::align_val_t al) {
-  auto ptr = malloc(count);
+#if defined(_WIN32) || defined(__CYGWIN__)
```

Comment:
you can make these macros or functions above that has one platform switch and then call them in these places to make this a little less eye-bleedy :)

Reply:
SG, I refactored a little. I also added a table to the PR description with tested configurations and known issues.